Aave ParaSwap Adapters

miguelmtzinf opened the pull request:
Based on the approach used for `ParaSwapDebtAdapter` by @bgd-labs, Aave Labs created a new version of the existing [ParaSwap adapters of V3](https://github.com/aave/aave-v3-periphery/tree/master/contracts/adapters/paraswap) with enhanced logic and usability.

This PR:
- Add adapter contracts for the following actions: repay with collateral, collateral swap and withdraw and swap.
- Add details about ParaSwap integration and security considerations.
- Reorder directory structure, including contracts and tests.
- Fix natspec docs and add clean-ups across contracts.

Looking forward to BGD's feedback on these contracts, and how we can collaborate together on this matter

miguelmtzinf committed:
feat: revamp of paraswap adapters (#12)

Co-authored-by: Parth Patel <parth4321patel@gmail.com>

sakulstra commented on foundry.toml:
```diff
@@ -1,7 +1,7 @@
-src = 'src'
-test = 'tests'
-script = 'scripts'
+src = 'src/contracts'
```
just curious, why does it matter?

miguelmtzinf replied:
per foundry docs, it references the path to contract sources. Not sure if a wrong config can bring any issue

sakulstra commented on scripts/psp.js:
```diff
@@ -60,11 +61,11 @@ function augustusFromAmountOffsetFromCalldata(calldata) {
       return 68; // 4 + 2 * 32
     case "0x3865bde6": // directCurveV1Swap
-      return 68; // 4 + 2 * 32
+      return 132; // 4 + 4 * 32
```
good, closes https://github.com/bgd-labs/aave-debt-swap/issues/26

sakulstra commented on scripts/psp.js:
```diff
@@ -91,17 +92,17 @@ const augustusToAmountOffsetFromCalldata = (calldata) => {
-  // distinguish between exactOut and exactInoutdMethod
-  const excludedMethod =
+  // distinguish between exactOut and exactInOutMethod
+  const preferredMethods =
```
preferred imo sounds wrong as they are not preferred, but exlusively used, would say in line with paraswap naming `includeContractMethods`

sakulstra commented on src/contracts/ParaSwapLiquiditySwapAdapterV2.sol:
```diff
@@ -0,0 +1,54 @@
+    // Intentionally left blank
+  }
+
+  /// @inheritdoc BaseParaSwapAdapter
```
shouldn't it be possible to abstact this into a AaveV2 and AaveV3 contract to just extend? seems a bit odd to implement this stuff for each adapter.

sakulstra commented on src/contracts/base/ParaSwapLiquiditySwapAdapter.sol:
```diff
@@ -0,0 +1,233 @@
+   * @notice Renews the asset allowance to the Aave Pool
+   * @param reserve The address of the asset
+   */
+  function renewAllowance(address reserve) public {
```
it doesn't really make sense for this to be public, does it?

sakulstra commented on src/contracts/base/ParaSwapLiquiditySwapAdapter.sol:
```diff
@@ -0,0 +1,233 @@
+      liquiditySwapParams.paraswapData,
+      IERC20Detailed(flashLoanAsset),
+      IERC20Detailed(liquiditySwapParams.newCollateralAsset),
+      flashLoanAmount - flashLoanPremium,
```
imo this should use the `liquiditySwapParams.collateralAmountToSwap` instead of calculating the value.
Technically it should be the same, but imo it's a bit weird to use a "different" value then the one supplied by the user.

sakulstra replied:
after some discussion with @parth-15 i agree now that it's fine. Thx for clarification.

parth-15 committed:
fix: rename variable in psp.js (#14)

parth-15 committed:
fix: move common function to base contract (#13)

* fix: move common function to base contract

* fix: make allowance function internal

* remove redundant function

miguelmtzinf commented:
Complicated to fix existing conflicts given that this PR performs major changes in files and directories structure.  @sakulstra lmk what you think

miguelmtzinf committed:
fix: Resolve conflicts agains BGD upstream remote main chain (#15)

* chore: bnb deploy script (#25)

* fix: resolve conflicts

miguelmtzinf committed:
Merge branch 'main' into fix/upstream-conflicts

miguelmtzinf committed:
Merge pull request #16 from aave/fix/upstream-conflicts

fix: Resolve merge conflicts with BGD upstream main

absis committed:
chore: add dependency-review action workflow

absis committed:
Merge pull request #20 from aave/francesc/devops-980-implement-dependency-review-action-on-aave-repos

chore: add Dependency Review GitHub Action

dependabot[bot] committed:
chore(deps): bump axios from 1.1.3 to 1.13.5

Bumps [axios](https://github.com/axios/axios) from 1.1.3 to 1.13.5.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.1.3...v1.13.5)

---
updated-dependencies:
- dependency-name: axios
  dependency-version: 1.13.5
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

alexmance committed:
Merge pull request #21 from aave/dependabot/npm_and_yarn/axios-1.13.5

chore(deps): bump axios from 1.1.3 to 1.13.5
